File: php/lib/steam/packets/S2A_LOGSTRING_Packet.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2010, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Packets
 */

require_once STEAM_CONDENSER_PATH . 'steam/packets/SteamPacket.php';

/**
 * The S2A_LOGSTRING packet type is used to transfer log messages.
 *
 * @package Steam Condenser (PHP)
 * @subpackage Packets
 */
class S2A_LOGSTRING_Packet extends SteamPacket
{

    /**
     * @var string The log message contained in this packet
     */
    private $message;

    /**
     * Creates a new log message packet
     *
     * @param string $data
     */
    public function __construct($data) {
        parent::__construct(SteamPacket::S2A_LOGSTRING_HEADER, $data);

        $this->contentData->getByte();
        $this->message = $this->contentData->getString();
    }

    /**
     * @return string The log message of this packet
     */
    public function getMessage() {
        return $this->message;
    }

}
?>

File: ruby/lib/steam/packets/s2a_logstring_packet.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2011, Sebastian Staudt

require 'steam/packets/steam_packet'

# This class represents a S2A_LOGSTRING packet used to transfer log messages
#
# @author Sebastian Staudt
class S2A_LOGSTRING_Packet

  include SteamPacket

  # Returns the log message contained in this packet
  #
  # @return [String] The log message
  attr_reader :message

  # Creates a new S2A_LOGSTRING object based on the given data
  #
  # @param [String] data The raw packet data sent by the server
  def initialize(data)
    super S2A_LOGSTRING_HEADER, data

    @content_data.byte
    @message = @content_data.string
  end

end
